src/test/kotlin/org/rust/cargo/project/model/impl/SyncToolWindowTest.kt

```diff
@@ -316,6 +318,79 @@ class SyncToolWindowTest : RsWithToolchainTestBase() {
         """)
     }
 
+    @MinRustcVersion("1.62.0")
+    @WithExperimentalFeatures(EVALUATE_BUILD_SCRIPTS)
+    fun `test Cargo continues build script evaluation after a compilation error in an independent crate`() {
+        if (SystemInfo.isWindows) return // It looks like error messages are a bit different on Windows
+        val project = buildProject {
+            toml("Cargo.toml", """
+                [workspace]
+                members = [
+                    "with-build-script",
+                    "with-proc-macro"
+                ]
+            """)
+
+            dir("with-build-script") {
+                toml("Cargo.toml", """
+                    [package]
+                    name = "with-build-script"
+                    version = "0.1.0"
+                """)
+
+                dir("src") {
+                    rust("lib.rs", """
+                        fn main() {}
+                    """)
+                }
+                rust("build.rs", """
+                    fn main() { panic!("Foo!"); }
+                """)
+            }
+
+            dir("with-proc-macro") {
+                toml("Cargo.toml", """
+                    [package]
+                    name = "with-proc-macro"
+                    version = "0.1.0"
+
+                    [lib]
+                    proc-macro = true
+                    path = "src/lib.rs"
+
+                    [[bin]]
+                    path = "src/main.rs"
+                    name = "with-proc-macro-bin"
+                """)
+
+                dir("src") {
+                    rust("main.rs", """
+                        fn main() {}
+                    """)
+                    rust("lib.rs", """
+                        fn foobar() { bar } // Compile error
+                    """)
+                }
+            }
+        }
+        checkSyncViewTree("""
+            -
+             -finished
+              -Sync ${project.root.name} project
+               Getting toolchain version
+               -Updating workspace info
+                -Build scripts evaluation
+                 Compiling with-build-script v0.1.0
+                 -Compiling with-proc-macro v0.1.0
+                  -lib.rs
+                   Cannot find value `bar` in this scope
+                 -${project.root.name}
```

Comment:
Hm, weird message structure. I suppose `Compiling with-build-script` and `Compiling with-proc-macro` should be on the same level
@mchernyavsky could you check it, please?